### PR TITLE
orjson support

### LINKED ISSUE #73: Make orjson as the default json serializer
[orjson](https://github.com/ijl/orjson) is a rust based json serializer and getting more popularity. I would like to change the default json serializer to `orjson`
* If `orjson` discovered, use `orjson
* if not discovered, use `json`

---

Here is the benchmark result. Smaller is better 👍 

## `json`

```
--- se-small ---
pyserde                                         0.088839, 0.087945, 0.089258, 0.089191, 0.088485
--- se-medium ---
pyserde                                         0.075264, 0.079170, 0.086694, 0.075400, 0.074962
--- de-small ---
pyserde                                         0.098418, 0.097855, 0.096926, 0.096757, 0.097403
--- de-medium ---
pyserde                                         0.094796, 0.098535, 0.094846, 0.094659, 0.094789
```

## `orjson`

```
--- se-small ---
pyserde                                         0.034388, 0.034202, 0.034044, 0.034112, 0.034198
--- se-medium ---
pyserde                                         0.039117, 0.038936, 0.038580, 0.038555, 0.038569
--- de-small ---
pyserde                                         0.055549, 0.055119, 0.054846, 0.055648, 0.054613
--- de-medium ---
pyserde                                         0.094554, 0.094573, 0.094484, 0.094366, 0.094432
```

### PR DESCRIPTION
**Pull request by kigawas**
Closes #73 

Notable change: `to_json` now returns ~`bytes` in lieu of `str`~ compact str

### CODE REVIEW

**Comment by kigawas** [comment]
@yukinarit You can review when you have time

**Comment by yukinarit** [comment]
@kmsquire Could you review?

File: serde/json.py [hunk 1]
```diff
@@ -1,50 +1,73 @@
...
+        return orjson.loads(s, **opts)
+
+except ImportError:
+    import json
+
+    def json_dumps(obj, **opts):
+        if "default" not in opts:
+            opts["default"] = encode_numpy
+        # compact output
+        return json.dumps(obj, ensure_ascii=False, separators=(",", ":"), **opts)
```
Comment (yukinarit) [inline serde/json.py:29]: Could you fix to allow users to pass own `ensure_ascii` and `separators` variables in `to_json` like this? 

```python
ensure_ascii = opts.pop("ensure_ascii", False)
separators = opts.pop("separators", (",", ":"))
return json.dumps(obj, ensure_ascii=ensure_ascii, separators=separators, **opts)
```

I got this error without it
```
TypeError: json.dumps() got multiple values for keyword argument 'separators
```

**Review by yukinarit — CHANGES_REQUESTED** [review]

**Review by yukinarit — APPROVED** [review]
LGTM!

**Comment by kmsquire** [comment]
Sorry not to review sooner--I've been pretty busy--but glad this is merged.  Cheers!

**Comment by yukinarit** [comment]
@kmsquire 
No worries, and thank you so much! If you find anything is missing for numpy support, please report 👍

**Pull request merged**